### PR TITLE
[REVIEW] Move quantiles.hpp files to legacy

### PR DESCRIPTION
**Pull request by cwharris**
Part of #2942

### CODE REVIEW

**Comment by harrism** [comment]
@cwharris group_quantiles is related to groupby. Porting it should probably should be a separate PR, since it will depend on #3297 and the sort-based groupby followup to that which @devavret will be handling.

**Comment by cwharris** [comment]
`meta.yaml` doesn't contain a reference to `quantiles`, afaict.

What needs to happen to benchmarks? Move to `legacy`, adjust `CMakeLists.txt`? I didn't see an example of other legacy benchmarks having been moved yet, so wasn't sure.

**Comment by cwharris** [comment]
got a little overexcited with `group_quantiles.hpp` :smile:

**Comment by harrism** [comment]
Hmmm, so everyone missed benchmarks? That's because they aren't compiled in CI. Grrr....

Just make sure compilation succeeds when you turn them on.

**Comment by harrism** [comment]
meta.yaml -- unfortunately there are a lot of things missing. You should add any headers that are in cpp/include to meta.yaml.  It is how we make sure the right files get installed.

**Comment by cwharris** [comment]
Regarding benchmarks, the transition guide only _explicitly_ mentions them here:
```
2. Update paths to `old.hpp`, `old.cpp`, and `old_tests.cpp` 
    - `cudf/cpp/CMakeLists.txt` 
    - `cudf/cpp/tests/CMakeLists.txt`
    - `cudf/conda/recipes/libcudf/meta.yaml`
    - Include paths (including code in `cudf/cpp/benchmarks`)
    - Cython include paths (see [Cython changes](#cython_changes))
```

**Comment by cwharris** [comment]
rerun tests

**Comment by harrism** [comment]
Of course, just update the benchmark includes so they still work with legacy. Porting the benchmarks to use the new APIs is part of the porting task.

File: cpp/include/cudf/legacy/io_types.hpp [hunk 1]
```diff
@@ -21,7 +21,7 @@
 #include <memory>
 #include <utility>
 
-#include <cudf/cudf.h>
+// #include <cudf/cudf.h>
```
Comment (harrism) [inline cpp/include/cudf/legacy/io_types.hpp]: Remove commented out code.

File: cpp/include/cudf/legacy/quantiles.hpp [hunk 2]
```diff
@@ -16,8 +16,8 @@
 
 #pragma once
 
-#include "cudf.h"
-#include "types.hpp"
+#include <cudf/cudf.h>
```
Comment (harrism) [inline cpp/include/cudf/legacy/quantiles.hpp]: Try to eliminate includes of cudf.h altogether. It just includes a whole lot of files. Change to only including the needed files.

**Pull request merged**